modules/github-oidc-role/main.tf

```diff
@@ -0,0 +1,54 @@
+locals {
+  enabled             = module.this.enabled
+  aws_canned_policies = [for arn in var.aws_iam_policies : arn if can(regex("^arn:aws:iam::aws:policy/", arn))]
+  aws_policies        = length(local.aws_canned_policies) > 0 ? local.aws_canned_policies : null
+  policy_document_map = {
+    "gitops"        = local.gitops_policy
+    "lambda_cicd"   = local.lambda_cicd_policy
+    "inline_policy" = one(module.iam_policy.*.json)
+  }
```

Comment:
This needs to be overridable/extensible. See example from `aws-team-roles`: https://github.com/cloudposse/terraform-aws-components/blob/1b338fe664e5debc5bbac30cfe42003f7458575a/modules/aws-team-roles/main.tf#L10-L16